watertap/property_models/ion_DSPMDE_prop_pack.py

```diff
@@ -0,0 +1,838 @@
+
+    CONFIG.declare("solute_list", ConfigValue(
+        domain=list,
+        description="List of solute species names"))
+    CONFIG.declare("stokes_radius_data", ConfigValue(
+        default={},
+        domain=dict,
+        description="Dict of solute species names and Stokes radius data"))
+    CONFIG.declare("diffusivity_data", ConfigValue(
+        default={},
+        domain=dict,
+        description="Dict of solute species names and bulk ion diffusivity data"))
+    CONFIG.declare("mw_data", ConfigValue(
+        default={},
+        domain=dict,
+        description="Dict of component names and molecular weight data"))
+    CONFIG.declare("density_data", ConfigValue(
+        default={},
+        domain=dict,
+        description="Dict of component names and component density data"))
+    CONFIG.declare("charge", ConfigValue(
+        default={},
+        domain=dict,
+        description="Ion charge"))
+    CONFIG.declare("activity_coefficient_model", ConfigValue(
+        default=ActivityCoefficientModel.ideal,
+        domain=In(ActivityCoefficientModel),
+        description="Activity coefficient model construction flag",
+        doc="""
+           Options to account for activity coefficient model.
+    
+           **default** - ``ActivityCoefficientModel.ideal``
+    
+       .. csv-table::
+           :header: "Configuration Options", "Description"
+    
+           "``ActivityCoefficientModel.ideal``", "Activity coefficients equal to 1 assuming ideal solution"
+           "``ActivityCoefficientModel.davies``", "Activity coefficients estimated via Davies model"
+       """))
+
+    def _init_param_data(self, data, default=None):
+        if default is None:
+            default = 1
+        config = getattr(self.config, data)
+        if len(config) != 0:
+            return extract_data(config)
+        else: #TODO only works if no data provided at all, but should have conditional that handles one or more missing vals
+            _log.warning(f"Missing initial configuration values for {data}. All values being arbitrarily set to {default}. "
+                         f"Ensure correct data values are assigned before solving.")
+            return default
+
+    def build(self):
+        '''
+        Callable method for Block construction.
+        '''
+        super(DSPMDEParameterData, self).build()
+
+        self._state_block_class = DSPMDEStateBlock
+
+        # components
+        self.H2O = Solvent()
+
+        for j in self.config.solute_list:
+            self.add_component(str(j), Solute())
+
+        # phases
+        self.Liq = LiquidPhase()
+
+        # reference
+        # Todo: enter any relevant references
+
+
+        # molecular weight
+        self.mw_comp = Param(
+            self.component_list,
+            mutable=True,
+            initialize=self._init_param_data('mw_data'),
+            units=pyunits.kg/pyunits.mol,
+            doc="Molecular weight kg/mol")
```

Comment:
As a general rule I avoid putting units in the doc string now that the component itself has units. This just duplicates information and adds a potential point where data can get out of synch,